### DISCUSSION TITLE
Switching charging from grid on and off

### DISCUSSION THREAD

**Discussion by LudgerMi** [wlcrs/huawei_solar#575] [Setting up the integration] [answered]
### Describe your feature request

The app SUN2000 offers in the section 'maintenance' the charging of the battery from the grid. I cannot find or I did not understand how to switch on and off charging from grid with the home assistant integration. I would like to request the feature controlling of charging from grid.
Background: During winter time I need to buy electrical power from the grid. I would like to use a provider who offers dynamic prices for electrical power. I know someone who developed a HA integration for the provider ‘tibber’: Getting the current price and starting charging of his EV.

### Proper usage

- [X] I confirm that this is not a bug report or support request
- [X] I confirm that this feature request is within the stated scope of the integration

**Comment by wlcrs** [comment, marked as answer]
You can use the services exposed by this integration to start a forced charge:

![SmartSelect_20231230_153902_Home Assistant](https://github.com/wlcrs/huawei_solar/assets/2150060/9bc368de-6676-4066-95f7-2951c5ac3724)

If you need to prevent your battery from discharging, you can set the maximum discharge power to 0 .

**Reply by LudgerMi** [reply]
Dear Thijs, thanks for support, it works now, I'm very satisfied! 
Please apologize for inconvenience. I saw the announcement of these services, but I did not realize the connection to charging from grid.
Regards  Ludger

**Reply by max10019679** [reply]
Excuse my ignorance but I cannot work out how to navigate to that forcible charge screen shown above. Please could you give me the directory to it?

**Reply by LudgerMi** [reply]
Select Developer tools and Services. To find the service enter forcible Charge
![image](https://github.com/wlcrs/huawei_solar/assets/129231013/8f790453-f0d3-4f46-be85-d135859082b8)

**Reply by LudgerMi** [reply]
To create a button, which you place on the dash board I created a script: Settings --> Automations & scenes:
![image](https://github.com/wlcrs/huawei_solar/assets/129231013/6364b25d-772a-4ffc-a0c9-60b3ef538f38)
Select Scripts:
![image](https://github.com/wlcrs/huawei_solar/assets/129231013/6414e921-e8e8-4a08-8047-bbee4ae221c4)
![image](https://github.com/wlcrs/huawei_solar/assets/129231013/8628387e-a2ff-4c7f-ac59-94a61622c98e)
Entity on Dash Board:
![image](https://github.com/wlcrs/huawei_solar/assets/129231013/fdb7332c-d93e-4df6-a5b3-5b1114730461)

**Reply by max10019679** [reply]
thanks very much for your help - extremely helpful. the one last part of the puzzle I'm struggling with is how to trigger the charge when another device does something. I want the home battery to grid charge whenever my EV charge starts charging, and stop charging when the EV charge stops?

**Comment by LudgerMi** [comment]
In order to avoid misunderstandings: In this thread there is a discussion of force charging the home battery from the grid (AC), not charging from the solar panels. 
Charging the home battery from the grid is a use case during periods, when you do not get sufficient power from the solar modules and you want to buy electrical power from a provider who offers lower prices during the night.
Please tell me know, which kind of charging you have in mind, then I can provide you further information.

**Reply by max10019679** [reply]
sorry for not being very clear. im trying to set up an automation whereby the huawei battery charges from grid when my EV charger starts charging (i have a dynamic energy tarrif specific to EV's). my EV charger is HACS integrated to HA and I know I'm using the right sensor on the EV charger (tested it with notifications). as per the screenshot below the automation I want is - 'when' the EV charger load goes above 500w 'and if' between 5:30am and 11:30pm 'then do'.... I cannot find the right action from the list list to trigger a charge from grid? the action i've selected in the screenshot doesn't work?
![automation snip](https://github.com/wlcrs/huawei_solar/assets/157296649/ba21ab87-a272-412c-9b25-98b8a4f046e5)

**Comment by LudgerMi** [comment]
Ok, now I got it. In the graphical Editor for automatisation you can directly call the service 'Huawei solar: Forcible charge'. Please see below the example: When I open my kitchen window then start charging from grit:
![grafik](https://github.com/wlcrs/huawei_solar/assets/129231013/1866fcb0-f149-48dd-8643-91c089706fb6)
Unfortunatelly, I cannot test it now, because we have a sunny day today and my batterie is already full. I let you know this evening.

**Reply by LudgerMi** [reply]
You can also select 'Forcible Charge of the battery to a certain State of Charge level'
![grafik](https://github.com/wlcrs/huawei_solar/assets/129231013/6bb289ad-e1fa-43d5-a365-64a9db3b2624)

**Reply by max10019679** [reply]
Thanks very much. You solved my problem

**Reply by LudgerMi** [reply]
Good to hear that it's working now. Thijs did create a very usefull integration. Unless not already done please think about a donation to  Thijs.
